modules/@apostrophecms/settings/index.js

```diff
@@ -0,0 +1,207 @@
+          )
+        ];
+        self.validateSettingsSchema(allSettingsFields, userSchema);
+
+        for (const field of userSchema) {
+          if (allSettingsFields.includes(field.name)) {
+            schema.push(field);
+          }
+        }
+        return schema;
+      },
+
+      // Validate that the fields configured in the settings module exist in the
+      // user schema and are not forbidden.
+      // XXX Temporary (Phase 2,3) extend the forbidden protected fields.
+      validateSettingsSchema(settingsFieldNames, userSchema) {
+        const forbiddenFields = [
+          'role',
+          // These will be allowed in Phase 2 and 3
+          'username',
+          'password',
+          'email'
+        ];
+        for (const name of settingsFieldNames) {
+          if (!userSchema.some(field => field.name === name)) {
+            throw new Error(`[@apostrophecms/settings] The field "${name}" is not a valid user field.`);
+          }
+          if (forbiddenFields.includes(name)) {
+            throw new Error(`[@apostrophecms/settings] The field "${name}" is forbidden.`);
+          }
+        }
+      },
+
+      // Get subform fields by subform name.
+      getSubformSchema(name) {
+        const subform = self.options.subforms[name];
+        if (!subform) {
+          throw self.apos.error('notfound', `Subform "${name}" not found.`);
+        }
+        return subform.fields.map(fieldName => {
+          return self.userSchema.find(field => field.name === fieldName);
+        });
+      },
+
+      getSubform(name) {
+        return self.subforms.find(subform => subform.name === name);
+      },
+
+      addToAdminBar() {
+        if (!self.hasSchema()) {
+          return;
+        }
+        self.apos.adminBar.add(
+          `${self.__meta.name}:settings`,
+          'apostrophe:settings',
+          false,
+          {
+            user: true
+          }
+        );
+      },
+
+      addSettingsModal() {
+        if (!self.hasSchema()) {
+          return;
+        }
+        self.apos.modal.add(
+          `${self.__meta.name}:settings`,
+          self.getComponentName('settingsModal', 'AposSettingsManager'),
+          { moduleName: self.__meta.name }
+        );
+      },
+
+      getBrowserData(req) {
+        return {
+          subforms: self.subforms,
+          action: self.action
+        };
+      }
+    };
+  },
+
+  restApiRoutes(self) {
+    return {
+      async getAll(req) {
+        if (!self.hasSchema() || !req.user) {
+          throw self.apos.error('notfound');
+        }
+        const user = await self.apos.user
+          .find(req, { _id: req.user._id })
+          .permission(false)
+          .toObject();
+
+        if (!user) {
+          throw self.apos.error('notfound');
+        }
+
+        const values = {
+          _id: user._id
+        };
+        for (const field of self.userSchema) {
+          values[field.name] = user[field.name];
+        }
+        return values;
```

Comment:
Slightly unusual for REST to return an object at this level, but it makes sense here. 👍